### PR TITLE
finished model, migration and seeder files. starting to work on route handlers and API docs

### PR DESCRIPTION
**Pull request by 360devin360**
created models for all tables, created tables with migration files and added starter code with seeders. preparing to start API docs route handlers next

### CODE REVIEW

**Review by 360devin360 — COMMENTED** [review]
great

**Comment by 360devin360** [comment]
Added Log in route handler with custom error handling

**Comment by 360devin360** [comment]
added get user information to /api/session which sends user: null if no user is logged in

**Comment by 360devin360** [comment]
added a route to get all spots based on results specified from API docs

File: backend/db/migrations/20240521091243-create-spot.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
+'use strict';
+/** @type {import('sequelize-cli').Migration} */
+let options = {}
```
Comment (AnthonyBronca) [inline backend/db/migrations/20240521091243-create-spot.js:3]: may need to add `options.tableName = <tablename>`

File: backend/db/migrations/20240521092742-create-review.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
+'use strict';
+/** @type {import('sequelize-cli').Migration} */
+
+let options = {}
```
Comment (AnthonyBronca) [inline backend/db/migrations/20240521092742-create-review.js:4]: may need to add `options.tableName = <tablename>`

File: backend/db/migrations/20240521104611-create-booking.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
+'use strict';
+/** @type {import('sequelize-cli').Migration} */
+
+let options = {}
```
Comment (AnthonyBronca) [inline backend/db/migrations/20240521104611-create-booking.js:4]: may need to add `options.tableName = <tablename>`

File: backend/db/migrations/20240521150858-create-review-images.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
+'use strict';
+/** @type {import('sequelize-cli').Migration} */
+let options  = {};
```
Comment (AnthonyBronca) [inline backend/db/migrations/20240521150858-create-review-images.js:3]: may need to add `options.tableName = <tablename>`

File: backend/db/migrations/20240521161915-create-spot-image.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
+'use strict';
+/** @type {import('sequelize-cli').Migration} */
+let options = {};
```
Comment (AnthonyBronca) [inline backend/db/migrations/20240521161915-create-spot-image.js:3]: may need to add `options.tableName = <tablename>`

File: backend/db/seeders/20240521162921-demo-spot-images.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+/** @type {import('sequelize-cli').Migration} */
+const {SpotImage} = require('../models')
+let options = {}
```
Comment (AnthonyBronca) [inline backend/db/seeders/20240521162921-demo-spot-images.js:5]: may need to add `options.tableName = <tablename>`

File: backend/routes/index.js [hunk 7]
```diff
@@ -2,14 +2,13 @@ const express = require('express');
...
 
-router.use('/api',apiRouter);
 
 // test route
-router.get('/hello/world',(req,res)=>{
-    res.cookie("XSRF-TOKEN",req.csrfToken());
-    res.send('Hello World!');
-});
-
+// router.get('/hello/world',(req,res)=>{
```
Comment (AnthonyBronca) [inline backend/routes/index.js:7]: You can delete these now

**Review by AnthonyBronca — COMMENTED** [review]
Loooking good!

**Review by irelius — COMMENTED** [review]
Looks good. Nice work

**Review by AnthonyBronca — APPROVED** [review]

**Review by irelius — APPROVED** [review]

**Pull request merged**